packages/react-native-web/src/exports/Modal/ModalAnimation.js

```diff
@@ -0,0 +1,160 @@
+/**
+ * Copyright (c) Nicolas Gallagher.
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import { useEffect, useState, useCallback, useMemo } from 'react';
+
+import StyleSheet from '../StyleSheet';
+import createElement from '../createElement';
+
+import type { ModalAnimationProps } from './types';
+
+
+const ANIMATION_DURATION = 300;
+
+function getAnimationStyle(animationType, visible) {
+  if (animationType === 'slide') {
+    return [
+      (visible ? styles.animatedIn : styles.animatedOut),
+      (visible ? styles.slideIn : styles.slideOut)
+    ]
+  }
+
+  if (animationType === 'fade') {
+    return [
+      (visible ? styles.animatedIn : styles.animatedOut),
+      (visible ? styles.fadeIn : styles.fadeOut)
+    ]
+  }
+
+  if (!visible) {
+    return styles.hidden;
+  }
+
+  return [];
+}
+
+function ModalAnimation(props: ModalAnimationProps) {
+  const {
+    children,
+    style,
+    animated,
+    animationType,
+    visible,
+    onShow,
+    onDismiss
+  } = props;
+
+  const [isRendering, setIsRendering] = useState(false);
+
+  // Resolve the "actual" animation type by checking the (somewhat deprecated)
+  // animated prop against the animationType prop
+  const computedAnimationType = useMemo(() => {
+    if (!animationType) {
+      if (animated) {
+        return 'slide';
+      } else {
+        return 'none';
+      }
+    }
+
+    return animationType;
+  }, [animationType, animated]);
+
+  const isAnimated = computedAnimationType !== 'none';
+
+  // Wrap the callbacks so we don't have to worry about them being null
+  const onShowCallback = useCallback(() => { if (onShow) { onShow(); } }, [onShow]);
+  const onDismissCallback = useCallback(() => { if (onDismiss) { onDismiss(); } }, [onDismiss])
```

Comment:
We don't need these extra `useCallback` wrappers, this can be moved inside the body of `animationEndCallback`